resources/hooks/local/pre-push

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+remote="$1"
+
+while read local_ref local_sha remote_ref remote_sha
+do
+    # Get the branch name
+    local_ref=$(basename $local_ref)
+    break
+done
+    
+# Fetch the GIT diff and format it as command input:
+DIFF=$(git diff --name-only "$remote"/"$local_ref"..HEAD --oneline)
```

Comment:
Grumphp internally base everything on the `git diff` that parse to find all the files involved in the commit/push so I dont' think that actually is the best choice.
My hope is to have that pr integrated and start to work on this edge-cases that during my development doesn't happened.